packages/hierarchy-builder/src/test/internal/operators/grouping/PropertiesGrouping.test.ts

```diff
@@ -0,0 +1,1482 @@
+
+        classStubs.stubEntityClass({
+          schemaName: "TestSchema",
+          className: "Class",
+          is: async () => false,
+        });
+        const propertyInfo: propertiesGrouping.PropertyGroupInfo = {
+          ecClass,
+          previousPropertiesGroupingInfo: [],
+          propertyGroup: { propertyName: "PropertyName" },
+        };
+        expect(await propertiesGrouping.createPropertyGroups(metadataProvider, nodes, propertyInfo, formatter)).to.deep.eq({
+          groupingType: "property",
+          grouped: [],
+          ungrouped: nodes,
+        });
+      });
+
+      it("doesn't group when nodes' properties and provided previous properties don't match", async () => {
+        const nodes = [
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x1" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [
+                    { propertyName: "PropertyName1", propertyValue: "PropertyValue1" },
+                    { propertyName: "PropertyName2", propertyValue: "PropertyValue2" },
+                  ],
+                },
+              },
+            },
+          }),
+        ];
+        const ecClass = { fullName: "TestSchema.Class" } as unknown as ECClass;
+        classStubs.stubEntityClass({
+          schemaName: "TestSchema",
+          className: "Class",
+          is: async () => true,
+        });
+        const propertyInfo: propertiesGrouping.PropertyGroupInfo = {
+          ecClass,
+          previousPropertiesGroupingInfo: [{ fullClassName: "TestSchema.Class", propertyGroup: { propertyName: "Other" } }],
+          propertyGroup: { propertyName: "PropertyName2" },
+        };
+        expect(await propertiesGrouping.createPropertyGroups(metadataProvider, nodes, propertyInfo, formatter)).to.deep.eq({
+          groupingType: "property",
+          grouped: [],
+          ungrouped: nodes,
+        });
+      });
+    });
+
+    it("doesn't group when byProperties isn't set", async () => {
+      const nodes = [
+        createTestProcessedInstanceNode({
+          key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x1" }] },
+        }),
+      ];
+      const ecClass = { fullName: "TestSchema.Class" } as unknown as ECClass;
+      const propertyInfo: propertiesGrouping.PropertyGroupInfo = {
+        ecClass,
+        previousPropertiesGroupingInfo: [],
+        propertyGroup: { propertyName: "PropertyName" },
+      };
+      expect(await propertiesGrouping.createPropertyGroups(metadataProvider, nodes, propertyInfo, formatter)).to.deep.eq({
+        groupingType: "property",
+        grouped: [],
+        ungrouped: nodes,
+      });
+    });
+
+    describe("Range isn't provided", async () => {
```

Comment:
this `describe ` block seems to be nested under `propertyInfo is different from nodes' byProperties processing params` - should it?

Reply:
I'd also change the block name to `value grouping`